drivers/flash/Kconfig

```diff
@@ -17,6 +17,56 @@ config FLASH_HAS_EX_OP
 	  This option is selected by drivers that support flash extended
 	  operations.
 
+config FLASH_HAS_EXPLICIT_ERASE
+	bool
+	help
+	  Device has does not do erase-on-write (erase-on-program, auto-erase
+	  on write) and requires explicit erase procedure to be programmed
+	  with random value, in place where it has already been programmed with
+	  some other value, as program can only change bits from erased-value
+	  to the opposite.
+	  All pure Flash devices are evolution of EEPROM where erase has
+	  been separated from write, EEPROM has erase-on-write, giving
+	  it advantage of higher write speeds at a cost of larger erase block.
+	  Note that explicit-erase capability does not warrants that
+	  write without erase is not allowed, taking the above restrictions,
+	  it only states that write of a random information will require
+	  erase.
+	  Erase is usually performed in pages, as we have chosen to name
+	  the unit in Zephyr, that may have different naming in device
+	  specifications, like pages, sectors or blocks, and may vary
+	  in size, depending how they are named by vendor.
+	  This option should be selected by drivers that serve devices with
+	  such characteristic and is used and may be used by users to provide
+	  paths in code that only serve such devices, and could be
+	  optimized-out by compiler in case where there is no such device in
+	  a system.
+
+config FLASH_HAS_NO_EXPLICIT_ERASE
+	bool
+	help
+	  Device does not require explicit erase before programming
+	  a new random value at any location that has been previously
+	  programmed with some other value.
+	  Note that the device may have erase-on-write (auto-erase),
+	  as for example in EEPROM devices, but may also have no erase
+	  at all.
+	  A device driver may still provide erase callback,
+	  especially if it is able to perform erase to accelerate
+	  further writes or is able to fill the area requested for
+	  erase, with single value, faster than consecutive writes
+	  that would be used to emulate erase.
+	  This option should be selected by drivers that serve
+	  devices with such characteristic and is used and may be
+	  used by users to provide paths in code that only serve
+	  such devices, and could be optimized-out by compiler in
+	  case where there is no such device in a system.
+	  This option should be selected for any device that
+	  can change storage bits, by write, from any value to opposite
+	  value at any time.
+	  When your driver sets this option you also need to set
+	  no_explicit_erase capability in your drivers flash_parameters.
```

Comment:
Not really sure why both of these flags are needed when you can use `CONFIG_FLASH_HAS_EXPLICIT_ERASE` and `!CONFIG_FLASH_HAS_EXPLICIT_ERASE`

Reply:
Because that is change in base of feature set. You can not tell whether you have devices that lack the feature once single device sets `CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE`. This means that you end up with code that has to support explicit erase devices whether they exist in the system or not. So on, for example nrfF54l15, using only the `CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE` you can go into code that will be good for that platform, but you have no way to throw out the other code, since there is no way to tell whether there are devices that do not have the feature.